keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/BlockDeviceConfig.kt

```diff
@@ -0,0 +1,195 @@
+package com.netflix.spinnaker.keel.ec2.resource
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.stereotype.Component
+
+
+@ConfigurationProperties(prefix = "keel.plugins.ec2.volumes.defaults")
+class VolumeDefaultConfiguration (
+  var volumeType : String = "gp2",
+  var deviceName : String = "/dev/sdb",
+  var deleteOnTermination : Boolean = true
+)
+
+data class BlockDevice(
+  val volumeType : String,
+  val deviceName : String,
+  val size : Int, // in GB
+  val deleteOnTermination : Boolean
+)
+
+
+/**
+ * Default config info for EBS block devices
+ *
+ * Note: this class only has config data for a subset of AWS instance types. In particularly, keel only knows
+ * about instance types where the clouddriver default is to mount a single EBS volume at "/dev/sdb"
+ */
+@Component
+@EnableConfigurationProperties(VolumeDefaultConfiguration::class)
+class BlockDeviceConfig(
+  private val config : VolumeDefaultConfiguration
+)
+ {
+
+  /**
+   * Get the list of block devices for this instance type
+   * For instance types unsupported by this class, keel doesn't pass EBS config info, which results in falling back to
+   * clouddriver's defaults
+   */
+  fun get(instanceType: String): List<BlockDevice>? =
+    volumeSizeGb(instanceType)?.let { size ->
+      listOf(
+        BlockDevice(volumeType=config.volumeType,
+          deviceName=config.deviceName,
+          size=size,
+          deleteOnTermination=config.deleteOnTermination)
+      )
+  }
+
+  /**
+   * Return the size of the EBS volume, in GB
+   *
+   * Copied over from:
```

Comment:
The redundancy here bothers me a bit. Did you consider exposing an API in clouddriver to return this data, which we could then just cache in keel?

Reply:
P.S. I'd be totally OK with making that an improvement for follow-up PRs.

Reply:
Yeah, this change felt like the quickest way to evaluate whether this actually resolves the issue for us. If we end up keeping this functionality, I'll coordinate with the clouddriver devs to see if we can design a better long-term solution.